File: epi_judge_cpp/minimum_waiting_time.cc
#include <vector>

#include "test_framework/generic_test.h"
using std::vector;

int MinimumTotalWaitingTime(vector<int> service_times) {
  
    if (service_times.empty())
        return 0;
    std::sort(service_times.begin(), service_times.end());
    int min_total_wait_time = 0;
    for (int i = 0; i < service_times.size()-1; i++) {
        min_total_wait_time += service_times[i] * (service_times.size() - (i + 1));
    
    }
    return min_total_wait_time;
}

int main(int argc, char* argv[]) {
  std::vector<std::string> args{argv + 1, argv + argc};
  std::vector<std::string> param_names{"service_times"};
  return GenericTestMain(args, "minimum_waiting_time.cc",
                         "minimum_waiting_time.tsv", &MinimumTotalWaitingTime,
                         DefaultComparator{}, param_names);
}

File: epi_judge_python_solutions/minimum_waiting_time.py
from typing import List

from test_framework import generic_test


def minimum_total_waiting_time(service_times: List[int]) -> int:

    # Sort the service times in increasing order.
    service_times.sort()
    return sum(
        service_time * (len(service_times) - (i + 1))
        for i, service_time in enumerate(service_times)
    )


def minimum_total_waiting_time_pythonic(service_times):
    return sum(
        remaining_queries * time
        for remaining_queries, time in enumerate(sorted(service_times)[::-1]))


if __name__ == '__main__':
    exit(
        generic_test.generic_test_main('minimum_waiting_time.py',
                                       'minimum_waiting_time.tsv',
                                       minimum_total_waiting_time))
